### PR TITLE
Xbrl nov archive

### PR DESCRIPTION
**Pull request by cmgosnell**
- [x] removed a few straggler warnings...
   - some columns  slipped into the small plants rename dict
   - some unmapped string categorizations from the small plants table
   - Note: There are nulls in that result in this warning: `[ WARNING] catalystcoop.pudl.transform.classes:465 fuel_type_from_header: Found 1 uncategorized values: {nan}`. I wasn't able to add a null into `na_categories` because the type checks expect strings. I thought about adding a `nullable` attribute into the param, but decided this was oos for this task. something to consider in the future though
- [x] converted the header/notes dropping into a `drop_invalid_rows`
   - i did this because it felt important to log how many header/notes where found in the transform step and figured we should use the standard way we drop records now. I moved the header/note param to the top because it wasn't dropping anything after the drop-no-data `drop_invalid_rows` param. This is expected bc the labeling process for labeling possible headers/notes looks for similar no-data records.
   - this required a tweak to the glue tests to add in empty columns for required columns before running the data through `drop_invalid_rows`. `drop_invalid_rows` was dropping the whole df if all of the `require_valid_columns` didn't exist in the df, so they now get adding for the glue tests. 
- [x] updated the archive
  - the new archive had upper and lower case `data_format` in the json so I had to tweak that.
- [x] ran the glue tests

### CODE REVIEW

**Review by aesharpe — COMMENTED** [review]
Looks good to me!

File: src/pudl/extract/ferc1.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ def get_file(self, year: int, filename: str):
         """Opens given ferc1 file from the corresponding archive."""
         if year not in self._cache:
             self._cache[year] = self.datastore.get_zipfile_resource(
-                "ferc1", year=year, data_format="DBF"
+                "ferc1", year=year, data_format="dbf"
```
Comment (zaneselvans) [inline src/pudl/extract/ferc1.py:269]: @zschira let's standardize on one or the other case in the archives here.

File: src/pudl/glue/ferc1_eia.py [hunk 2]
```diff
@@ -203,6 +203,36 @@ def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
...
+            for param in self.params.drop_invalid_rows
+            if param.required_valid_cols
+        ]
+        required_valid_cols = pudl.helpers.dedupe_n_flatten_list_of_lists(
+            list_of_lists_of_required_valid_cols
+        )
+        if required_valid_cols:
+            missing_required_cols = [
+                col for col in required_valid_cols if col not in df
+            ]
```
Comment (zaneselvans) [inline src/pudl/glue/ferc1_eia.py:226]: In general what you're really doing here is a set operation that answers the question "What elements are in `required_valid_cols` but not in `df.columns`?", which you can figure out by "subtracting" all of `df.columns` from `required_valid_cols` using `set.difference()`

```py
missing_required_cols = set(required_valid_cols).difference(df.columns)
```

Using lists preserves ordering (which isn't typically meaningful here) and will also preserve duplicate list elements, which isn't appropriate.

It's probably doing the right thing here, but I think it's better to use sets for set operations.

**Review by zaneselvans — COMMENTED** [review]
No show stoppers but some questions / comments.

File: src/pudl/extract/ferc1.py (hunk 1 above)
Comment (cmgosnell) [reply src/pudl/extract/ferc1.py:269]: i added [an issue](https://github.com/catalyst-cooperative/pudl-archiver/issues/12) in maybe the right repo

File: src/pudl/transform/ferc1.py [hunk 3]
```diff
@@ -1410,13 +1410,10 @@ def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
             .pipe(self.associate_notes_with_values)
             .pipe(self.spot_fix_rows)
             .pipe(self.drop_invalid_rows)
+            # Now remove the row_type columns because we've already moved totals to a
+            # different column
+            .drop(columns=["row_type"])
         )
-        # Remove headers and note rows now that the relevant information has been
-        # extracted.
-        df = df[(df["row_type"] != "header") & (df["row_type"] != "note")].copy()
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:1416]: i created a `drop_invalid_row` param to do this work

File: src/pudl/transform/params/ferc1.py [hunk 4]
```diff
@@ -2454,12 +2473,6 @@
                     "row_number": "row_number",
                     "row_seq": "row_seq",
                     "spplmnt_num": "spplmnt_num",
-                    "begin_yr_bal": "starting_balance",
-                    "addition": "additions",
-                    "retirements": "retirements",
-                    "adjustments": "adjustments",
-                    "transfers": "transfers",
-                    "yr_end_bal": "ending_balance",
```
[the remark this answers is not in the document]
Comment (cmgosnell) [reply src/pudl/transform/params/ferc1.py:2462]: yeah they snuck with some merge conflicts. begone.

File: src/pudl/transform/ferc1.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:1416]: Ah okay so it's just happening in that other step. Got it.

File: src/pudl/transform/params/ferc1.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (zaneselvans) [reply src/pudl/transform/params/ferc1.py:2462]: sneaky bastards.

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**